command/agent/cache/cacheboltdb/bolt.go

```diff
@@ -207,7 +205,12 @@ func (b *BoltStorage) GetByType(ctx context.Context, indexType string) ([][]byte
 			return fmt.Errorf("bucket %q not found", b.rootBucket)
 		}
 		top.Bucket([]byte(indexType)).ForEach(func(id, cipherText []byte) error {
-			plainText, err := b.keyMgr.Decrypt(ctx, cipherText, []byte(b.aad))
+			var blob wrapping.EncryptedBlobInfo
+			if err := proto.Unmarshal(cipherText, &blob); err != nil {
+				return err
+			}
+
+			plainText, err := b.wrapper.Decrypt(ctx, &blob, []byte(b.aad))
```

Comment:
Looks like there's two cases of this unmarshal and decrypt block, so it may be worth moving into it's own function?